pkg/daemon/cleanup/cleanup.go

```diff
@@ -142,8 +147,20 @@ func mergeServices(s1, s2 []servicescm.Service) []servicescm.Service {
 
 }
 
+// mergeCMData combines the data from two ConfigMaps into a single data
+func mergeCMData(data1, data2 map[string]string) map[string]string {
+	mergedData := make(map[string]string)
+	for key, value := range data1 {
+		mergedData[key] = value
+	}
+	for key, value := range data2 {
+		mergedData[key] = value
+	}
+	return mergedData
+}
```

Comment:
Don't you need to merge the services and envvars independently, and then create a new CM from the merged values? Doing this seems like it will just take everything (files, services, etc) from mergedData, as the keys will be the same.